_posts/blog/2016-10-08-vector_drawables_optimisation.md

```diff
@@ -0,0 +1,188 @@
+---
+layout: post
+title: "Optimizing the Performance of Vector Drawables"
+description: This is how vector drawables work and how to optimize them
+modified:
+categories: blog
+author: florina_muntenescu
+excerpt: You have replaced all image resources in your app with VectorDrawables - your APK is smaller and your images look better. But are you really using them correctly? Here are three mistakes that you might be making and how to fix them, right now. Because &#35;PERFMATTERS.
+tags: [Android, UI, Vector Drawables, perfmatters]
+image:
+date: 2016-10-08T00:39:55-04:00
+---
+
+
+While some mobile platforms have been supporting vector graphics for a while, Android only began doing this natively starting with API Level 21 and with the help of the Support Library 23.2.0 for pre-Lollipop devices. By replacing you PNG image resources with VectorDrawables, your APK size decreases considerably and most of all, your images look good, independent of the resolution of the device used.
+
+
+The system will try to redraw your Activity every 16ms so it can reach the <a href="https://www.youtube.com/watch?v=CaMTIgxCSqU">60fps</a> target. So this means that the duration of every `onDraw` method is extremely important.
+
+
+When used incorrectly, VectorDrawables can affect the performance of your app, because the drawing can take a long time. Here are three mistakes that you might be making, but can be solved easily, to ensure that you are really improving the performance of your app, with the help of VectorDrawables.
+
+
+## Understanding The Internals Of VectorDrawables
+
+
+Vector graphics use geometrical shapes to describe graphical elements. The vector graphics are rendered at runtime. The automatic rendering at pixel density gives smoothness to the graphics, regardless of the device capabilities. So your images won’t be downscaled or upscaled, looking stretched or pixelated but they will be always perfectly drawn for your screen size.
+Vector drawables allow the representation of images (e.g. icons, UI elements) based on XML vector graphics.
```

Comment:
either delete the last sentence or put it at the beginning of the paragraph - it does not provide new information